datafusion/expr/src/logical_plan/extension.rs

```diff
@@ -195,6 +195,10 @@ pub trait UserDefinedLogicalNode: fmt::Debug + Send + Sync {
     /// directly because it must remain object safe.
     fn dyn_eq(&self, other: &dyn UserDefinedLogicalNode) -> bool;
     fn dyn_ord(&self, other: &dyn UserDefinedLogicalNode) -> Option<Ordering>;
+
+    /// Indicates to the optimizer if its safe to push a limit down past
+    /// this extension node
+    fn allows_limit_to_inputs(&self) -> bool;
```

Comment:
I recommend renaming this method `supports_limit_pushdown` to be consistent with the `ExecutionPlan` method:


https://docs.rs/datafusion/latest/datafusion/physical_plan/trait.ExecutionPlan.html#method.supports_limit_pushdown

Reply:
Also, if we added a default implementation, it would ease migration:

```suggestion
    fn allows_limit_to_inputs(&self) -> bool { false } 
```